### PR TITLE
[eas-cli] Add support for roll back to embedded updates

### PR DESCRIPTION
**Pull request by wschurman**
# Checklist

- [ ] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

This adds a new command, `eas update:roll-back-to-embedded` (name up for discussion), that creates a "roll back to embedded" update.

Server implementation: https://github.com/expo/universe/pull/11720
Client implementations: https://github.com/expo/expo/pull/21652, https://github.com/expo/expo/pull/21256
Protocol spec: https://www.notion.so/expo/Expo-Updates-directives-in-multipart-manifests-2dee4deac86241a4873ee2fdb3d83d9c

# How

I considered making this a flag, but it seems the trend is to have these be a separate command (see `update:republish`).

1. Create command
2. Factor out common code from index
3. Add test

# Test Plan

Run test.

Run:
```
~/expo/eas-cli/packages/eas-cli/bin/run update:roll-back-to-embedded --private-key-path keys/private-key.pem
```
(to check that a republish is created in staging and that it is signed)

### CODE REVIEW

**Comment by wschurman** [comment]
/changelog-entry new-feature Add support for roll back to embedded updates

File: packages/eas-cli/src/commands/update/roll-back-to-embedded.ts [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+  branchName?: string;
+  channelName?: string;
+  updateMessage?: string;
+  privateKeyPath?: string;
+  json: boolean;
+  nonInteractive: boolean;
+};
+
+export default class UpdateRollBackToEmbedded extends EasCommand {
+  static override description = 'roll back to the embedded update';
```
Comment (dsokal) [inline packages/eas-cli/src/commands/update/roll-back-to-embedded.ts:69]: Not really related to the PR, but can you please update the wording in https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/commands/update/republish.ts#L59? (rollback -> roll back)

File: packages/eas-cli/src/commands/update/roll-back-to-embedded.ts [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.DynamicProjectConfig,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    const { flags: rawFlags } = await this.parse(UpdateRollBackToEmbedded);
+    const flags = this.sanitizeFlags(rawFlags);
```
Comment (dsokal) [inline packages/eas-cli/src/commands/update/roll-back-to-embedded.ts:109]:
```suggestion
```

File: packages/eas-cli/src/project/publish.ts [hunk 3]
```diff
@@ -453,3 +467,186 @@ export function isUploadedAssetCountAboveWarningThreshold(
...
+          )}`,
+        }),
+        paginatedQueryOptions,
+      });
+      branchName = branch.name;
+    } catch {
+      // unable to select a branch (network error or no branches for project)
+      ({ name: branchName } = await promptAsync({
+        type: 'text',
+        name: 'name',
```
Comment (dsokal) [inline packages/eas-cli/src/project/publish.ts:525]: Redundant rename?
```suggestion
      ({ branchName } = await promptAsync({
        type: 'text',
        name: 'branchName',
```

File: packages/eas-cli/src/utils/code-signing.ts [hunk 4]
```diff
@@ -181,3 +189,24 @@ export function checkManifestBodyAgainstUpdateInfoGroup(
     assertAssetParity(correspondingManifestResponseBodyAssetJSON, nullthrows(partialManifestAsset));
   }
 }
+
+function areSetsEqual<T>(a: Set<T>, b: Set<T>): boolean {
+  return a.size === b.size && [...a].every(value => b.has(value));
+}
```
Comment (dsokal) [inline packages/eas-cli/src/utils/code-signing.ts:195]: Maybe move to `expodash` + add a test?

**Review by dsokal — COMMENTED** [review]
I don't have much context on this so my review is mostly nitpicking.

**Review by douglowder — APPROVED** [review]

**Pull request merged**